### PR TITLE
I/6546: Implement remove column util

### PR DESCRIPTION
**Pull request by jodator**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Feature: Introduce `TableUtils.removeColumns()` method. Closes ckeditor/ckeditor5#6546. Closes ckeditor/ckeditor5#6439.

---

### Additional information

- Unfortunately this PR only fixes one of the recent table bugs.

### CODE REVIEW

File: tests/commands/removecolumncommand.js [hunk 1]
```diff
@@ -440,11 +440,11 @@ describe( 'RemoveColumnCommand', () => {
...
-				[ { rowspan: 2, contents: '[]00' } ]
+				[ '[]00' ]
 			] ) );
 		} );
 
-		it( 'should work property if the rowspan is in the first column (#3)', () => {
+		it( 'should work property if the rowspan is in the first column (the cell with rowspan is selected)', () => {
 			setData( model, modelTable( [
 				[ { rowspan: 2, contents: '00[]' }, '01' ],
 				[ '10' ]
```
Comment (niegowski) [inline tests/commands/removecolumncommand.js:450]: Should be `11`.

File: tests/tableutils.js [hunk 2]
```diff
@@ -960,4 +960,248 @@ describe( 'TableUtils', () => {
...
+			} );
+
+			it( 'should decrease colspan of table cells from previous column', () => {
+				setData( model, modelTable( [
+					[ { colspan: 4, contents: '00' }, '03' ],
+					[ { colspan: 3, contents: '10' }, '13' ],
+					[ { colspan: 2, contents: '20' }, '22', '23' ],
+					[ '30', { colspan: 2, contents: '31' }, '33' ],
+					[ '40', '41', '42', '43' ]
+				] ) );
```
Comment (niegowski) [inline tests/tableutils.js:1021]: This table seems invalid:
```
+----+----+----+----+----+
| 00                | 03 |
+----+----+----+----+----+
| 10           | 13 |
+----+----+----+----+
| 20      | 22 | 23 |
+----+----+----+----+
| 30 | 31      | 33 |
+----+----+----+----+
| 40 | 41 | 42 | 43 |
+----+----+----+----+
```

File: tests/tableutils.js [hunk 3]
```diff
@@ -960,4 +960,248 @@ describe( 'TableUtils', () => {
...
+					[ '40', '41', '43' ]
+
+				] ) );
+			} );
+
+			it( 'should decrease colspan of cells that are on removed column', () => {
+				setData( model, modelTable( [
+					[ { colspan: 3, contents: '00' }, '03' ],
+					[ { colspan: 2, contents: '10' }, '13' ],
+					[ '20', '21', '22', '23' ]
```
Comment (niegowski) [inline tests/tableutils.js:1039]: Seems invalid table:
```
+----+----+----+----+
| 00           | 03 |
+----+----+----+----+
| 10      | 13 |
+----+----+----+----+
| 20 | 21 | 22 | 23 |
+----+----+----+----+
```

File: tests/tableutils.js [hunk 4]
```diff
@@ -960,4 +960,248 @@ describe( 'TableUtils', () => {
...
+					[ { colspan: 2, contents: '00' }, '03' ],
+					[ '10', '13' ],
+					[ '21', '22', '23' ]
+				] ) );
+			} );
+
+			it( 'should remove column with rowspan (first column)', () => {
+				setData( model, modelTable( [
+					[ { rowspan: 2, contents: '00' }, '01' ],
+					[ '10' ]
```
Comment (niegowski) [inline tests/tableutils.js:1054]: Should be `11`.

File: tests/commands/removecolumncommand.js [hunk 5]
```diff
@@ -367,37 +367,37 @@ describe( 'RemoveColumnCommand', () => {
 
 		it( 'should decrease colspan of table cells from previous column', () => {
 			setData( model, modelTable( [
-				[ { colspan: 4, contents: '00' }, '03' ],
-				[ { colspan: 3, contents: '10' }, '13' ],
-				[ { colspan: 2, contents: '20' }, '22[]', '23' ],
-				[ '30', { colspan: 2, contents: '31' }, '33' ],
-				[ '40', '41', '42', '43' ]
+				[ { colspan: 4, contents: '00' }, '04' ],
+				[ { colspan: 3, contents: '10' }, '14' ],
```
Comment (niegowski) [inline tests/commands/removecolumncommand.js:371]: Missing `13`.

File: tests/tableutils.js [hunk 6]
```diff
@@ -1013,52 +1013,52 @@ describe( 'TableUtils', () => {
 
 			it( 'should decrease colspan of table cells from previous column', () => {
 				setData( model, modelTable( [
-					[ { colspan: 4, contents: '00' }, '03' ],
-					[ { colspan: 3, contents: '10' }, '13' ],
-					[ { colspan: 2, contents: '20' }, '22', '23' ],
-					[ '30', { colspan: 2, contents: '31' }, '33' ],
-					[ '40', '41', '42', '43' ]
+					[ { colspan: 4, contents: '00' }, '04' ],
+					[ { colspan: 3, contents: '10' }, '14' ],
```
Comment (niegowski) [inline tests/tableutils.js:1017]: Missing `13`.

File: src/tableutils.js [hunk 7]
```diff
@@ -307,6 +307,66 @@ export default class TableUtils extends Plugin {
 		} );
 	}
 
+	/**
+	 * Removes columns from the given `table`.
+	 *
+	 * This method re-calculates the table geometry including `colspan` attribute of table cells overlapping removed columns
```
Comment (oleq) [inline src/tableutils.js:313]:
```suggestion
	 * This method re-calculates the table geometry including the `colspan` attribute of table cells overlapping removed columns
```

**Pull request merged**